sql: prevent overflow of field position in split_part

spilchen opened the pull request:
Previously, the field position argument to `split_part` could cause an integer overflow when passed a value of math.MinInt64. This happened because we attempted to negate the value with `-1 * field`, which fails for MinInt64 since its negation cannot be represented in a signed 64-bit integer.

This change updates the bounds check to properly handle this edge case without overflowing.

Fixes #145888

Epic: none
Release note (bug fix): Fixed an integer overflow in the split_part function when using extremely negative field positions like math.MinInt64.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146271)
<!-- Reviewable:end -->

spilchen committed:
sql: prevent overflow of field position in split_part

Previously, the field position argument to `split_part` could cause an
integer overflow when passed a value of math.MinInt64. This happened
because we attempted to negate the value with `-1 * field`, which fails
for MinInt64 since its negation cannot be represented in a signed
64-bit integer.

This change updates the bounds check to properly handle this edge case
without overflowing.

Fixes #145888

Epic: none
Release note (bug fix): Fixed an integer overflow in the split_part
function when using extremely negative field positions like
math.MinInt64.

spilchen commented:
TFTR!

bors r+

spilchen commented:
blathers backport 25.2
